File: src/main/java/io/github/liquidTruffle/parser/ast/nodes/TextNode.java
package io.github.liquidTruffle.parser.ast.nodes;

import io.github.liquidTruffle.parser.ast.AstNode;
import com.oracle.truffle.api.frame.VirtualFrame;
import com.oracle.truffle.api.nodes.NodeInfo;

@NodeInfo(description = "Represents literal text content in Liquid templates")
public class TextNode extends AstNode {
    private final String text;
    
    public TextNode(String text) {
        this.text = text;
    }
    
    @Override
    public String executeGeneric(VirtualFrame frame) {
        return text;
    }
    
    // Expose text for testing purposes
    public String getTextContent() {
        return this.text;
    }
}

File: src/main/kotlin/io/github/liquidTruffle/ast/nodes/TextNode.kt
package io.github.liquidTruffle.ast.nodes

import io.github.liquidTruffle.ast.AstNode
import com.oracle.truffle.api.frame.VirtualFrame
import com.oracle.truffle.api.nodes.NodeInfo

@NodeInfo(description = "Represents literal text content in Liquid templates")
class TextNode(private val text: String) : AstNode() {
	override fun executeGeneric(frame: VirtualFrame): String = text
	
	// Expose text for testing purposes
	val textContent: String get() = this.text
}
